.github/workflows/test.yaml

```diff
@@ -13,15 +13,44 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
-      - name: Install cairo compiler
-        run: |
-          curl -L -o $ARCHIVE_NAME $ARCHIVE_LINK
-          tar -xvf $ARCHIVE_NAME
+      - name: Cache Scarb binary
+        uses: actions/cache@v2
+        with:
+          path: ~/.scarb
+          key: scarb-${{ env.SCARB_VERSION }}-${{ runner.os }}
+          restore-keys: scarb-${{ env.SCARB_VERSION }}-
 
-      - name: Check Cairo compiler version and add to path
+      - name: Scarb Setup
         run: |
-          $(pwd)/cairo/bin/cairo-compile --version
-          echo "$(pwd)/cairo/bin" >> $GITHUB_PATH
+          if [[ ! -f $HOME/.scarb/bin/scarb ]]; then
+            wget https://github.com/software-mansion/scarb/releases/download/${{ env.SCARB_VERSION }}/scarb-${{ env.SCARB_VERSION }}-x86_64-unknown-linux-gnu.tar.gz
+            mkdir $HOME/.scarb
+            tar -xvzf scarb-${{ env.SCARB_VERSION }}-x86_64-unknown-linux-gnu.tar.gz -C $HOME/.scarb --strip-components=1
+          fi
+          echo "$HOME/.scarb/bin" >> $GITHUB_PATH
+      
+      
+      - name: Scarb Build
+        run: scarb build
+
+      - name: Cache Protostar installation
+        uses: actions/cache@v2
+        with:
+          path: ~/.protostar
+          key: protostar-${{ env.PROTOSTAR_VERSION }}-${{ runner.os }}
+          restore-keys: protostar-${{ env.PROTOSTAR_VERSION }}-
 
+      - name: Protostar Setup
+        run: |
+          if [[ ! -d $HOME/.protostar/dist/protostar ]]; then
+            curl -L https://raw.githubusercontent.com/software-mansion/protostar/master/install.sh | bash -s -- -v $PROTOSTAR_VERSION
+          fi
+          echo "export PATH=\"\$PATH:$HOME/.protostar/dist/protostar\"" >> $GITHUB_PATH
+          source $GITHUB_PATH          
```

Comment:
why do we need this line? it wasn't needed before for the Cairo compiler?

Reply:
updated